app/assets/javascripts/discourse/initializers/topic-footer-buttons.js.es6

```diff
@@ -1,29 +1,112 @@
+import showModal from "discourse/lib/show-modal";
 import { registerTopicFooterButton } from "discourse/lib/register-topic-footer-button";
 
 export default {
   name: "topic-footer-buttons",
 
   initialize() {
```

Comment:
This method has become very long. Maybe it should be broken up into one method per footer button?

Reply:
Yes agreed, is that ok if I do it in a separate PR ?